File: Backend/src/main/java/com/espumablanca/backend/application/usecase/ProfileService.java
package com.espumablanca.backend.application.usecase;

import com.espumablanca.backend.domain.model.dto.request.profile.ProfileRequest;
import com.espumablanca.backend.domain.model.dto.request.profile.ProfileUpdatePasswordRequest;
import com.espumablanca.backend.domain.model.dto.response.profile.ProfileDto;

public interface ProfileService {
    ProfileDto getProfile(Long id);

    ProfileDto updateProfile(Long id, ProfileRequest request);

    ProfileDto updatePassword(Long id, ProfileUpdatePasswordRequest request);
}

File: Frontend/src/app/core/services/api/profile.service.ts
import { HttpClient } from '@angular/common/http';
import { inject, Injectable } from '@angular/core';
import { environment } from '../../../../environments/environment.development';
import { ProfileDto } from '../../models/profile/response/profile-dto.model';
import { ProfileRequest } from '../../models/profile/request/profile-request.model';
import { ProfileUpdatePasswordRequest } from '../../models/profile/request/profile-update-password-request.model';

@Injectable({
  providedIn: 'root',
})
export class ProfileService {
  private readonly endpoint = environment.apiUrl + '/protected/profile';
  private readonly http = inject(HttpClient);

  getProfile(id: number) {
    return this.http.get<ProfileDto>(this.endpoint + '/' + id);
  }

  update(id: number, request: ProfileRequest) {
    return this.http.put<ProfileDto>(this.endpoint + '/' + id, request);
  }

  updatePassword(id: number, request: ProfileUpdatePasswordRequest) {
    return this.http.put<ProfileDto>(
      this.endpoint + '/' + id + '/update/password',
      request
    );
  }
}
